core/state/access_list.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type accessList struct {
+	addresses map[common.Address]int
+	slots     []map[common.Hash]struct{}
+}
+
+// Contains returns true if the address is in the access list.
+func (al *accessList) ContainsAddr(address common.Address) bool {
+	_, ok := al.addresses[address]
+	return ok
+}
+
+// Contains returns whether the (address, slot) is present.
+func (al *accessList) Contains(address common.Address, slot common.Hash) (addressPresent bool, slotPresent bool) {
```

Comment:
I've vote to expand to `ContainsSlot`, since we have another contains method too and it's not obviout what simply `Contains` is (e.g. I myself would have made Contains operate on accounts, not on slots since those are higher level concepts).

Reply:
Aah, this checks both the address and the slot. Hmm, misunderstood based on the doc. Perhaps:

```
// Contains checks if a slot within an account is present in the access list, returning
// separate flags for the account and slot.
```

Reply:
![Screenshot_2020-10-19 EIP-2929 Gas cost increases for state access opcodes + YoloV2 by holiman · Pull Request #21509 · ethe](https://user-images.githubusercontent.com/142290/96496052-06be9d00-1249-11eb-9203-e887f64d3ae5.png)

:rofl: